backend/packages/worker/src/worker/parsing/parser.py

```
@@ -0,0 +1,50 @@
+
+class DocumentParser(Protocol):
+    def parse(self, doc_id: str, file_path: str) -> ParsedArtifact:
+        ...
+
+class DummyPDFParser:
+    """A dummy parser implementation for testing text extraction and parsing."""
+    def parse(self, doc_id: str, file_path: str) -> ParsedArtifact:
+        # In a real scenario, this would use PyMuPDF or pdfplumber
+        # This dummy implementation just creates an artifact from the string content of a file
+        try:
+            with open(file_path, 'r') as f:
```

Ruff (0.15.11)

[warning] Unnecessary mode argument

Remove mode argument

(UP015)